### PR TITLE
P8 skeleton viewer

### PR DESCRIPTION
**Pull request by Pryme8**
Initial PR to get the ball rolling here, this just gets what I need in place and does not break anything with the older stuff.

Was told to break my work into smaller PR's so here we go!

### CODE REVIEW

**Comment by deltakosh** [comment]
you need to run "npm run build" before doing a PR to check for documentation and linting

File: src/Debug/boneWeightShader.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    selectedWeight: Nullable<number[]>|DynamicTexture
+    colorMap: Nullable<number[]>|DynamicTexture
+}*/
+
+export class BoneWeightShader{    
+    public name?: Nullable<string>    
+    public skeleton: Skeleton
+    public mesh: AbstractMesh
+    private _scene: Scene  
+    //private _renderingMode: number
```
Comment (deltakosh) [inline src/Debug/boneWeightShader.ts:31]: need to be removed

File: src/Debug/boneWeightShader.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    constructor(
+        public options: Partial<IBoneWeightOptions> = {}
+    ){
+
+        this.options.renderingMode = options.renderingMode || BoneWeightShader.MODE_SELECTED_WEIGHT 
+        this.options.selectedColorKnots = options.selectedColorKnots || [0.25, Color3.Blue(), 0.5, Color3.Green(), 0.75, Color3.Yellow(), 1, Color3.Red()],
+        this.options.mapColorKnots = options.mapColorKnots || [0.25, Color3.Purple(), 0.5, (Color3.Red().add(Color3.Yellow())).scale(0.5), 0.75, Color3.Green(), 1, Color3.Teal()] 
+
+        this._scene = this.skeleton.getScene()
+        //this._renderingMode = this.options.renderingMode
```
Comment (deltakosh) [inline src/Debug/boneWeightShader.ts:43]: same

File: src/Debug/boneWeightShader.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                grad.addColorStop(pos, col)
+            }
+        }
+        ctx.fillStyle = grad
+        ctx.fillRect(0, 0, this.bones.length, 0)
+        dt.update()    
+    }
+
+    build(){
+        //let vxString: string;
```
Comment (deltakosh) [inline src/Debug/boneWeightShader.ts:66]: same

File: src/Debug/boneWeightShader.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        this.options.renderingMode = options.renderingMode || BoneWeightShader.MODE_SELECTED_WEIGHT 
+        this.options.selectedColorKnots = options.selectedColorKnots || [0.25, Color3.Blue(), 0.5, Color3.Green(), 0.75, Color3.Yellow(), 1, Color3.Red()],
+        this.options.mapColorKnots = options.mapColorKnots || [0.25, Color3.Purple(), 0.5, (Color3.Red().add(Color3.Yellow())).scale(0.5), 0.75, Color3.Green(), 1, Color3.Teal()] 
+
+        this._scene = this.skeleton.getScene()
+        //this._renderingMode = this.options.renderingMode      
+    }
+
+    _prep(){
```
Comment (deltakosh) [inline src/Debug/boneWeightShader.ts:46]: This is not yet called right?

Comment (Pryme8) [reply src/Debug/boneWeightShader.ts:46]: No I was just getting it in place.

File: src/Debug/boneWeightShader.ts (hunk 1 above)
Comment (Pryme8) [reply src/Debug/boneWeightShader.ts:31]: What needs to be removed?

Comment (Pryme8) [reply src/Debug/boneWeightShader.ts:31]: Ohh I get you!  Yeah, for sure Ill have all comments stripped here soon.

My phone was weird and was showing the entire commit and I got confused.

File: src/Debug/boneWeightShader.ts (hunk 4 above)
Comment (Pryme8) [reply src/Debug/boneWeightShader.ts:46]: the weight shader is not called by anything yet but will here in the next few commits.

**Comment by Pryme8** [comment]
I am having a ton of trouble now getting my local environment to work and I have no clue why.
Please see this thread:
https://forum.babylonjs.com/t/monoco-edtior-is-not-letting-me-build-or-npm-run-start/12842/2

I need help.

**Comment by Pryme8** [comment]
I am closing this one and starting a new one.  My whole branch got messed up somehow and even reverting it did nothing.
The solution was to start back from step one and just copy and past my code over into a new branch.  I will be doing the PR from there.

**Pull request closed without merging**